### PR TITLE
MGMT-8193 - Remove download_image logic from assisted_service_api.py and use infra_env.download_url to download it

### PR DESCRIPTION
**Pull request by eliorerz**
From [MGMT-8193](https://issues.redhat.com/browse/MGMT-8193):

> Currently whenever we need to download the discovery iso during a CI test we're using some API client which assumes we're using the download endpoint on assisted service.
> 
> When we move to the image service, that download url will no longer work and our tests will fail even though the flow for users would be unchanged.
> 
> We should be relying on the same url we present to users when we run CI (the one in the cluster image info and infraEnv).
> 

Deprecated methods in this PR:

1. `assisted_service_client._download`
2. `assisted_service_client.download_infraenv_image`
3. `assisted_service_client.download_image`

/cc @carbonin @osherdp 

FYI @lalon4 @nshidlin

### CODE REVIEW

**Comment by nshidlin** [comment]
asked on the jira, but asking here again. is the `/v2/infra-envs/{infra_env_id}` path being removed all together?

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 1]
```diff
@@ -167,22 +167,23 @@ def generate_and_download_infra_env(
         ssh_key=None,
         ignition_info=None,
         proxy=None,
-    ):
+    ) -> Path:
+        if self._config.is_static_ip:
```
Comment (nshidlin) [inline discovery-infra/test_infra/helper_classes/cluster.py:171]: strange behavior to except `static_network_config` and then override it. should be `self._config.is_static_ip and not static_network_config`

Comment (eliorerz) [reply discovery-infra/test_infra/helper_classes/cluster.py:171]: Yeah, probably missed that. It was late 😅

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 2]
```diff
@@ -751,17 +752,14 @@ def _ha_not_none(self):
...
-            static_network_config = None
-
-        self.generate_and_download_infra_env(
-            iso_download_path=iso_download_path or self._config.iso_download_path,
-            iso_image_type=self._config.iso_image_type,
-            static_network_config=static_network_config,
-        )
+    def download_image(self, iso_download_path: str = None) -> Path:
+        if self._infra_env is None:
+            log.warning("No infra_env found. Generating infra_env and downloading ISO")
```
Comment (carbonin) [inline discovery-infra/test_infra/helper_classes/cluster.py:757]: I don't think finding or not finding the infraEnv should change this behavior.
It's still valid to call generate for v1 and update the infraEnv in v2.

File: discovery-infra/test_infra/helper_classes/cluster.py [hunk 3]
```diff
@@ -167,22 +167,23 @@ def generate_and_download_infra_env(
         ssh_key=None,
         ignition_info=None,
         proxy=None,
-    ):
+    ) -> Path:
+        if self._config.is_static_ip and static_network_config is None:
```
Comment (carbonin) [inline discovery-infra/test_infra/helper_classes/cluster.py:171]: What does this have to do with changing how we download the image?

File: discovery-infra/test_infra/assisted_service_api.py [hunk 4]
```diff
@@ -166,6 +166,12 @@ def get_infra_env_by_cluster_id(self, cluster_id: str) -> List[models.infra_env.
 
     @classmethod
     def _download(cls, response: HTTPResponse, file_path: str, verify_file_size=False) -> None:
+        warnings.warn(
```
Comment (carbonin) [inline discovery-infra/test_infra/assisted_service_api.py:169]: Instead of deprecating these, does it make sense to just change their implementation?

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 3 above)
Comment (eliorerz) [reply discovery-infra/test_infra/helper_classes/cluster.py:171]: I moved that from `download_image` to here.
It's needed for generating the `infra_env`

File: discovery-infra/test_infra/assisted_service_api.py (hunk 4 above)
Comment (eliorerz) [reply discovery-infra/test_infra/assisted_service_api.py:169]: I thought about it but we will anyway need to change the function signature (to pass infra_env or infra_env arguments).
Also, it's not really needed here, now we can download image directly from `infra_env`  using `infra_env.download_image`.
I would just delete it if no one was using it

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 2 above)
Comment (carbonin) [reply discovery-infra/test_infra/helper_classes/cluster.py:757]: Not sure I fully understand, but I don't think it can hurt :man_shrugging:

**Comment by carbonin** [comment]
Looks good, but did you want that last test to pass? Do you think those failures are caused by this change?

**Comment by eliorerz** [comment]
> Looks good, but did you want that last test to pass? Do you think those failures are caused by this change?

Thanks, I don't think that issue is related..
see https://search.ci.openshift.org/?search=+Timeout+of+360+seconds+expired+waiting+for+Waiting+for+Ips&maxAge=168h&context=1&type=bug%2Bjunit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

File: discovery-infra/test_infra/helper_classes/cluster.py (hunk 2 above)
Comment (eliorerz) [reply discovery-infra/test_infra/helper_classes/cluster.py:757]: It does generate the image. As far as I know, creating infra_env generates also an ISO
